pyclam/criterion.py

```diff
@@ -337,3 +337,36 @@ def _select(root: Cluster, upper: float, lower: float, grace_depth: int) -> Set[
             inactive = {child for cluster in inactive for child in cluster.children}
 
         return selected
+
+
+class Labels(SelectionCriterion):
+    def __init__(self, labels: Dict[int, int]):
+        """ Uses a given dictionary of index -> label to select optimal clusters.
+
+        During a BFT of the tree, when a cluster has an overwhelming majority of points with the same label,
```

Comment:
What is an overwhelming majority? Can we parameterize that?

Reply:
It was already parameterized in self.threshold. I added an explanation to the doc-string.